### ISSUE TITLE
Allow for authenticating GitHub Rest calls with GITHUB_TOKEN

### ISSUE THREAD

**Issue by Raphiiko** [ZhongRuoyu/setup-llvm#7]
I've been running into rate limits fairly quickly with this action:
![image](https://github.com/ZhongRuoyu/setup-llvm/assets/111654848/d626ebf1-aa6c-4802-b2bf-e0f51b23b4d7)

As it seems to happen on a call to GitHub's API, there should be a way to improve this by allowing users to authenticate by giving it their token: https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-github_token-in-github-actions

**Comment by ZhongRuoyu** [comment]
Thanks for the issue; this should be resolved by #8. You should now be able to add a `github-token` input to the action like this (`${{ github.token }}` is already the default so you don't have to):

```yaml
- uses: ZhongRuoyu/setup-llvm@v0
  with:
    github-token: ${{ github.token }}
```

**Comment by Raphiiko** [comment]
Fantastic, thanks a lot for the quick add!
